kong/plugins/prometheus/exporter.lua

```diff
@@ -31,6 +31,65 @@ package.loaded['prometheus_resty_counter'] = require("resty.counter")
 local kong_subsystem = ngx.config.subsystem
 local http_subsystem = kong_subsystem == "http"
 
+
+-- should we introduce a way to know if a plugin is configured or not?
+local instance_update_register, has_instance do
+  local plugin_name = "prometheus"
+  local instance_count = 0
+
+
+  local function initialize_instance_counters()
+    instance_count = 0
+    for plugin, err in kong.db.plugins:each() do
+      if err then
+        kong.log.crit("could not obtain list of plugins: ", err)
+        return nil, err
+      end
+
+      if plugin.name == plugin_name then
+        instance_count = instance_count + 1
+        return
+      end
+    end
+  end
+
+
+  function instance_update_register()
+    local worker_events = kong.worker_events
+
+    -- in case we have prometheus plugin configured
+    initialize_instance_counters()
+
+    -- DB-less or DP.
+    if kong.configuration.database == "off" then
+      worker_events.register(function(data)
+        if data.entity.name ~= plugin_name then
+          return
+        end
+
+        local operation = data.operation
+        if operation == "create" then
+          instance_count = instance_count + 1
+        elseif operation == "delete" then
+          instance_count = instance_count - 1
+        end
+      end, "crud", "plugins")
```

Comment:
Let's not do this, this is not a reliable way of maintaining the `instance_count` as worker events could be lost and bring `instance_count` into a forever wrong state.

We should simply store a cached value called `"prometheus_enabled"` or so and `invalidate` it when `crud` happens and `data.entity.name == "prometheus"`. Then we use `cache.get` and a callback to determine if there is any enabled Prometheus instance anywhere.

Reply:
done.